process_hierarchy.py

```diff
@@ -0,0 +1,478 @@
+            case "pi":
+                process_record(cursor, record, hierarchy["pi"])
+            case "openshift-project":
+                process_record(cursor, record, hierarchy["project"])
+            case "openstack-project":
+                process_record(cursor, record, hierarchy["project"])
+                record["parent_id"] = record["id"]
+                process_record(cursor, record, hierarchy["cloud-project"])
+    return 1
+
+
+def process_coldfront_data(cursor, hierarchy):
+    """this processes the coldfront data (pi and project/cloud-project
+        into the hierarchy
+    )"""
+    coldfront_data = get_data_from_coldfront()
+    for record in coldfront_data:
+        # have to assume that the PI is active if it is coming from cold-front
+        pi_rec = {
+            "type": "pi",
+            "name": record["project"]["pi"],
+            "display_name": record["project"]["pi"],
+            "status": "Active",
+        }
+        pi_id = find_hierarchy_id(pi_rec["name"], hierarchy["pi"])
+        if not pi_id:
+            # need to assign unknown as field of science and univeristy
+            pi_rec["parent_id"] = find_hierarchy_id(
+                "unknown", hierarchy["field-of-science"]
+            )
+
+        process_record(cursor, pi_rec, hierarchy["pi"])
+
+        project_rec = {
+            "name": record["attributes"]["Allocated Project ID"],
+            "display_name": record["attributes"]["Allocated Project Name"],
+            "parent_id": record["project"]["pi"],
+            "status": record["status"],
+        }
+
+        if (
+            record["resource"]["name"] == "NERC"
+            and record["resource"]["resource_type"] == "OpenShift"
+        ):
+            project_rec["type"] = "openshift-project"
+            process_record(cursor, project_rec, hierarchy["project"])
+        elif (
+            record["resource"]["name"] == "NERC"
+            and record["resource"]["resource_type"] == "OpenStack"
+        ):
+            project_rec["type"] = "openstack-project"
+            process_record(cursor, project_rec, hierarchy["project"])
+            project_rec["parent_id"] = record["id"]
+            process_record(cursor, project_rec, hierarchy["cloud-project"])
+        else:
+            print("Unknown project record type")
+            pprint.pprint(record)
+
+
+def main():
+    """the main function"""
+    config = {"host": "localhost", "xdmod_password": "pass", "admin_password": "pass"}
+
+    # try:
+    #    with open("/etc/xdmod/xdmod_init.json", "r", encoding="utf-8") as file:
+    #        config = json.load(file)["database"]
+    # except IOError:
+    #    print("Ensure the xdmod-init.json file is in /etc/xdmod/xdmod_init.json")
+
+    cnx = moc_db_helper_functions.connect_to_db(config)
+    cursor = cnx.cursor(dictionary=True)
+
+    # This is here for development
+    if moc_db_helper_functions.db_exist(cursor, "hierarchy_db"):
```

Comment:
I would strongly recommend putting this either in a separate file, or behind a specific argument when calling the scripts and not part of the main execution.

Reply:
yeah, this has outlived it's purpose.